clinical-domain-agent/src/main/java/care/smith/fts/cda/DefaultTransferProcessRunner.java

```diff
@@ -21,28 +28,97 @@ public class DefaultTransferProcessRunner implements TransferProcessRunner {
 
   private final Map<String, TransferProcessInstance> instances = new ConcurrentHashMap<>();
 
-  @Value("${transfer.sendConcurrency:32}")
-  private int sendConcurrency = 32;
+  private final Queue<TransferProcessInstance> queued = new ConcurrentLinkedQueue<>();
+  private final TransferProcessRunnerConfig config;
+
+  public DefaultTransferProcessRunner(@Autowired TransferProcessRunnerConfig config) {
+    this.config = config;
+  }
 
   @Override
   public String start(TransferProcessDefinition process, List<String> pids) {
     var processId = UUID.randomUUID().toString();
     log.info("Run process with processId: {}", processId);
-    log.debug("Using a sendConcurrency of {}", sendConcurrency);
-    TransferProcessInstance transferProcessInstance =
-        new TransferProcessInstance(process, processId);
-    transferProcessInstance.execute(pids);
-    instances.put(processId, transferProcessInstance);
+    var transferProcessInstance = new TransferProcessInstance(process, processId, pids);
+
+    startOrQueue(processId, transferProcessInstance);
+
     return processId;
   }
 
+  private void startOrQueue(String processId, TransferProcessInstance transferProcessInstance) {
+    removeOldProcesses();
+    instances.compute(
+        processId,
+        (k, v) -> {
+          if (runningInstances() < config.maxConcurrentProcesses) {
+            transferProcessInstance.execute();
+            return transferProcessInstance;
+          } else {
+            queued.add(transferProcessInstance);
+            return null;
+          }
+        });
+  }
+
+  private long runningInstances() {
+    return instances.values().stream().filter(TransferProcessInstance::isRunning).count();
+  }
+
+  private void removeOldProcesses() {
+    instances.values().stream()
+        .filter(p -> nonNull(p.status.get().finishedAt()))
+        .filter(
+            p ->
+                p.status
+                    .get()
+                    .finishedAt()
+                    .isBefore(
+                        LocalDateTime.now().minus(Duration.ofSeconds(config.processTtlSeconds))))
```

Comment:
Using `now` instead of a Clock is difficult for tests, because you can't control the time.